packages/compass-preferences-model/src/provider.ts

```diff
@@ -1,18 +1,68 @@
 import { createContext, useContext } from 'react';
-import type { PreferencesAccess } from './';
-import { defaultPreferencesInstance } from '.';
+import type {
+  AllPreferences,
+  PreferenceStateInformation,
+  PreferencesAccess,
+  UserConfigurablePreferences,
+} from './';
+import { getDefaultsForAllPreferences } from './preferences';
 export { usePreference, withPreferences } from './react';
 export { capMaxTimeMSAtPreferenceLimit } from './maxtimems';
 
+export class SimplePreferenceService implements PreferencesAccess {
```

Comment:
Can we rename this to something like 
```suggestion
export class MockPreferenceService implements PreferencesAccess {
```

? It doesn't really implement the interface from a semantic point of view, so we should make clear that this is _only_ for testing

Reply:
Yea I am good with the suggested name. But just to state the intention - this is not just for testing but also for cases like Data Explorer where we need some preferences overrides but we don't ever modify them.

Reply:
Addressed this also in [77631fd](https://github.com/mongodb-js/compass/pull/5356/commits/77631fdcce2bab017575e2434c91798d5f19950f)